packages/request-client.js/test/index.test.ts

```diff
@@ -1149,6 +1149,108 @@ describe('request-client.js', () => {
     });
   });
 
+  describe('handle in-memory request', () => {
+    let requestNetwork: RequestNetwork;
+    let spyPersistTransaction: jest.Mock;
+    let mockServer: SetupServer;
+
+    beforeAll(() => {
+      spyPersistTransaction = jest.fn();
+
+      mockServer = setupServer(
+        http.post('*/persistTransaction', ({ request }) => {
+          if (!request.headers.get(config.requestClientVersionHeader)) {
+            throw new Error('Missing version header');
+          }
+          return HttpResponse.json(spyPersistTransaction());
+        }),
+        http.get('*/getTransactionsByChannelId', () =>
+          HttpResponse.json({
+            result: { transactions: [TestData.timestampedTransactionWithoutPaymentInfo] },
+          }),
+        ),
+        http.post('*/ipfsAdd', () => HttpResponse.json({})),
+        http.get('*/getConfirmedTransaction', () => HttpResponse.json({ result: {} })),
+      );
+      mockServer.listen({ onUnhandledRequest: 'bypass' });
+    });
+
+    beforeEach(() => {
+      spyPersistTransaction.mockReturnValue({});
```

Comment:
Doesn't it make sense to move it to `beforeAll` with the rest of the logic?

Reply:
I think so :+1:

Reply:
Done :white_check_mark: